Flaky test "test_records_order"

From https://github.com/ecolabdata/ecospheres-isomorphe/actions/runs/15084961112/job/42407012889 : 
```
================================== FAILURES ===================================
______________________________ test_records_order ______________________________
gn_client = <isomorphe.geonetwork.GeonetworkClientV3 object at 0x7fc0945355e0>
clean_md_fixtures = [Fixture(uuid='c375f298-bbc4-455b-98ad-f65a20b2e0a9', name='datara--c375f298-bbc4-455b-98ad-f65a20b2e0a9', content=b'<...md:LI_Lineage>\n         </gmd:lineage>\n      </gmd:DQ_DataQuality>\n  </gmd:dataQualityInfo>\n</gmd:MD_Metadata>\n')]
    def test_records_order(gn_client: GeonetworkClient, clean_md_fixtures: list[Fixture]):
        """Records should be ordered by changeDate asc"""
        records = gn_client.get_records()
        assert len(records) >= 2
    
        def get_change_date_from_record(record: Record) -> str:
            r = gn_client.get_record(record.uuid)
            return r.xpath(XPATH_ISO_DATE_STAMP, namespaces=r.nsmap)[0]
    
>       assert records == sorted(records, key=get_change_date_from_record)
E       AssertionError: assert [Record(uuid=...lished=False)] == [Record(uuid=...lished=False)]
E         
E         At index 0 diff: Record(uuid='c375f298-bbc4-455b-98ad-f65a20b2e0a9', title='Copy of record PPRN inondation Commune de SAINT VALLIER 26333 approuvé le 30/11/2012 created at 2024-09-06 10:40:28', md_type=<MetadataType.METADATA: 'n'>, state=None, published=False) != Record(uuid='04301749-df52-4c60-a640-87948241ac67', title='PPRN : Zonages réglementaires des plans de prévention des risques naturels inondation de CHARMES-SUR-RHONE (Ardèche)', md_type=<MetadataType.METADATA: 'n'>, state=None, published=False)
E         
E         Full diff:
E           [
E         +     Record(
E         +         uuid='c375f298-bbc4-455b-98ad-f65a20b2e0a9',
E         +         title='Copy of record PPRN inondation Commune de SAINT VALLIER 26333 '
E         +               'approuvé le 30/11/2012 created at 2024-09-06 10:40:28',
E         +         md_type=<MetadataType.METADATA: 'n'>,
E         +         state=None,
E         +         published=False,
E         +     ),
E               Record(
E                   uuid='04301749-df52-4c60-a640-87948241ac67',
E                   title='PPRN : Zonages réglementaires des plans de prévention des '
E                         'risques naturels inondation de CHARMES-SUR-RHONE (Ardèche)',
E                   md_type=<MetadataType.METADATA: 'n'>,
E                   state=None,
E                   published=False,
E               ),
E         -     Record(
E         -         uuid='c375f298-bbc4-455b-98ad-f65a20b2e0a9',
E         -         title='Copy of record PPRN inondation Commune de SAINT VALLIER 26333 '
E         -               'approuvé le 30/11/2012 created at 2024-09-06 10:40:28',
E         -         md_type=<MetadataType.METADATA: 'n'>,
E         -         state=None,
E         -         published=False,
E         -     ),
E           ]
tests/test_gn_client.py:73: AssertionError
```

Tried a few runs locally => can't reproduce.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -39,4 +39,4 @@ jobs:
 
       - name: Run tests
         run: |
-          pytest tests/
+          pytest -raP tests/

--- a/isomorphe/geonetwork.py
+++ b/isomorphe/geonetwork.py
@@ -198,18 +198,23 @@ class GeonetworkClient:
         """
         pass
 
-    def get_record(self, uuid: str) -> etree._ElementTree:
+    def get_record(self, uuid: str, query: dict[str, Any] | None = None) -> etree._ElementTree:
         log.debug(f"Processing record: {uuid}")
+        params = {
+            "addSchemaLocation": "true",  # FIXME: needed?
+            "increasePopularity": "false",
+            "withInfo": "false",
+            "attachment": "false",
+            "approved": "false",  # only relevant when workflow is enabled
+        }
+        if query:
+            params |= dict(
+                m(v) if (m := self.QUERY_MAPPINGS.get(k)) else (k, v) for k, v in query.items()
+            )
         r = self.session.get(
             f"{self.api}/records/{uuid}/formatters/xml",
             headers={"Accept": "application/xml"},
-            params={
-                "addSchemaLocation": "true",  # FIXME: needed?
-                "increasePopularity": "false",
-                "withInfo": "false",
-                "attachment": "false",
-                "approved": "false",  # only relevant when workflow is enabled
-            },
+            params=params,
         )
         r.raise_for_status()
         return etree.fromstring(r.content, parser=None)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -17,6 +17,7 @@ GN_TEST_USER: Final = "admin"
 GN_TEST_PASSWORD: Final = "admin"
 
 XPATH_ISO_DATE_STAMP = "/gmd:MD_Metadata/gmd:dateStamp/gco:DateTime/text()"
+XPATH_GN_DATE_STAMP = "/gmd:MD_Metadata/geonet:info/changeDate/text()"
 
 
 @dataclass(kw_only=True)
@@ -90,6 +91,7 @@ def seed_fixtures(gn_client: GeonetworkClient, group_fixture: int) -> list[Fixtu
                 group=None,
                 uuid_processing="NOTHING",
             )
+            sleep(1)  # make sure each record has a different database changeDate
             fixtures.append(Fixture(uuid=uuid, name=fixture.stem, content=content))
 
     total_records = gn_client.get_records()

--- a/tests/test_gn_client.py
+++ b/tests/test_gn_client.py
@@ -3,7 +3,7 @@ from pathlib import Path
 
 import pytest
 import requests_mock
-from conftest import XPATH_ISO_DATE_STAMP, Fixture
+from conftest import XPATH_GN_DATE_STAMP, Fixture
 
 from isomorphe.geonetwork import (
     GeonetworkClient,
@@ -67,8 +67,8 @@ def test_records_order(gn_client: GeonetworkClient, clean_md_fixtures: list[Fixt
     assert len(records) >= 2
 
     def get_change_date_from_record(record: Record) -> str:
-        r = gn_client.get_record(record.uuid)
-        return r.xpath(XPATH_ISO_DATE_STAMP, namespaces=r.nsmap)[0]
+        r = gn_client.get_record(record.uuid, query={"withInfo": "true"})
+        return r.xpath(XPATH_GN_DATE_STAMP, namespaces=r.nsmap)[0]
 
     assert records == sorted(records, key=get_change_date_from_record)
 

